src/libraries/System.Private.CoreLib/src/System/SR.cs

```diff
@@ -13,48 +13,19 @@ namespace System
 {
     internal static partial class SR
     {
-        // This method is used to decide if we need to append the exception message parameters to the message when calling SR.Format.
-        // by default it returns false.
-        // Native code generators can replace the value this returns based on user input at the time of native code generation.
-        // Marked as NoInlining because if this is used in an AoT compiled app that is not compiled into a single file, the user
-        // could compile each module with a different setting for this. We want to make sure there's a consistent behavior
-        // that doesn't depend on which native module this method got inlined into.
-        [MethodImpl(MethodImplOptions.NoInlining)]
-        private static bool UsingResourceKeys()
-        {
-            return false;
-        }
+        private static readonly object _lock = new object();
+        private static List<string>? _currentlyLoading;
+        private static int _infinitelyRecursingCount;
+        private static bool _resourceManagerInited;
 
         // Needed for debugger integration
         internal static string? GetResourceString(string resourceKey)
         {
             return GetResourceString(resourceKey, string.Empty);
         }
 
-        internal static string GetResourceString(string resourceKey, string? defaultString)
-        {
-            if (UsingResourceKeys())
-                return defaultString ?? resourceKey;
-
-            string? resourceString = null;
-            try { resourceString = InternalGetResourceString(resourceKey); }
-            catch (MissingManifestResourceException) { }
-
-            if (defaultString != null && resourceKey.Equals(resourceString, StringComparison.Ordinal))
-            {
-                return defaultString;
-            }
-
-            return resourceString!; // only null if missing resource
-        }
-
-        private static readonly object _lock = new object();
-        private static List<string>? _currentlyLoading;
-        private static int _infinitelyRecursingCount;
-        private static bool _resourceManagerInited = false;
-
         [PreserveDependency(".cctor()", "System.Resources.ResourceManager")]
-        private static string? InternalGetResourceString(string? key)
+        private static string? InternalGetResourceString(string key)
         {
```

Comment:
Should the below `string.IsNullOrEmpty` check be changed to just be a `Debug.Assert(!string.IsNullOrEmpty(key));` or are there valid situations where null might be passed in?  It's unclear to me from the comment below, but it seems like either we should put back the `?` or change the below to just be an assert.

Reply:
This is called with "" from https://github.com/dotnet/runtime/blob/build/src/libraries/System.Private.CoreLib/src/System/SR.cs#L24 which I'm not sure what is about as it is existing code.

> we should put back the ?

We could but the method is called only from methods which have it without `?`

Reply:
> This is called with ""

If there's an existing call site that would trigger the assert, then either the existing assert is bogus or the call site is buggy ;)

Reply:
Code updated to have better nullable annotation